File: backend/app/models/billing.py
from pydantic import BaseModel, Field
from typing import Optional

class BillingInDB(BaseModel):
    """Cấu trúc dữ liệu hóa đơn/thanh toán lưu trữ vật lý trong MongoDB"""
    billing_code: str = Field(..., description="Mã hóa đơn duy nhất (VD: INV-2026-001)")
    order_code: str = Field(..., description="Mã đơn hàng tham chiếu")
    
    amount_total: float = Field(..., description="Tổng số tiền cần thanh toán")
    currency: str = Field(default="VND")
    
    payment_method: str = Field(
        default="COD", 
        description="Phương thức thanh toán: COD, VNPAY, BANK_TRANSFER"
    )
    payment_status: str = Field(
        default="UNPAID", 
        description="Trạng thái: UNPAID, PENDING, PAID, FAILED, REFUNDED"
    )
    
    # Lưu mã giao dịch đối soát từ VNPay hoặc Ngân hàng trả về
    transaction_id: Optional[str] = None 
    
    # Thông tin người trả tiền (lưu dạng dict)
    payer_info: dict 
    
    # Metadata thời gian
    created_at: str
    updated_at: str
    paid_at: Optional[str] = None
    created_by: Optional[str] = None

File: backend/internal/billing/entity/billing.go
package entity

import "time"

type Billing struct {
	ID            int64      `json:"id" db:"id"`
	BillingCode   string     `json:"billing_code" db:"billing_code"`
	OrderCode     string     `json:"order_code" db:"order_code"`
	AmountTotal   float64    `json:"amount_total" db:"amount_total"`
	Currency      string     `json:"currency" db:"currency"`
	PaymentMethod string     `json:"payment_method" db:"payment_method"`
	PaymentStatus string     `json:"payment_status" db:"payment_status"`
	TransactionID string     `json:"transaction_id" db:"transaction_id"`
	PayerName     string     `json:"payer_name" db:"payer_name"`
	PayerPhone    string     `json:"payer_phone" db:"payer_phone"`
	PayerEmail    string     `json:"payer_email" db:"payer_email"`
	CreatedAt     time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at" db:"updated_at"`
	PaidAt        *time.Time `json:"paid_at" db:"paid_at"`
	CreatedBy     string     `json:"created_by" db:"created_by"`
}
